packages/graphql-language-service-server/src/languages.ts

```diff
@@ -0,0 +1,66 @@
+import { Position, Range } from 'graphql-language-service-utils';
+import { makeExtractTagsFromSource, GraphQLSource } from './languageUtils';
+
+export type LanguageConfig = {
+  readonly extensions: ReadonlyArray<string>;
+  readonly fileFilter: (content: string) => boolean;
+  readonly extractSources: (content: string) => ReadonlyArray<GraphQLSource>;
+};
+
+/**
+ * Filters and other regexps used later on.
+ */
+const jsFileFilter = new RegExp(/graphql`|graphql\.experimental`|gql`/);
+const jsGraphQLTagsRegexp = new RegExp(
+  /(?<=(graphql|gql|graphql\.experimental)`)[\s\S]+?(?=`)/g,
+);
+
+// This removes interpolation in JS strings, like ` ${someInterpolationHere}`
+const jsRemoveInterpolationRegexp = new RegExp(/([\s]+\${)[\s\S]+?(})/g);
+
+const jsExtractor = makeExtractTagsFromSource(jsGraphQLTagsRegexp);
+
+const reasonFileFilterRegexp = new RegExp(/(\[%(graphql|relay\.))/);
+const reasonGraphQLTagsRegexp = new RegExp(
+  /(?<=\[%(graphql|relay\.\w*)[\s\S]*{\|)[.\s\S]+?(?=\|})/g,
+);
+const reasonExtractor = makeExtractTagsFromSource(reasonGraphQLTagsRegexp);
+
+const jsLanguages: LanguageConfig = {
+  extensions: ['js', 'jsx', 'ts', 'tsx'],
+  fileFilter: (content: string) => jsFileFilter.test(content),
+  extractSources: (content: string) =>
+    jsExtractor(content).map(source => ({
+      ...source,
+      template: source.template.replace(jsRemoveInterpolationRegexp, ''),
+    })),
+};
+
+const graphql: LanguageConfig = {
+  extensions: ['graphql'],
```

Comment:
folks will want to add the other extension i think. on second thought from what i've said before, its fine to add `.gql` here as a catch all, as the spec never refers to it and always refers to .graphql files, so officially we do not endorse that extension but its there for the language server ok fine, haha

Reply:
I'll add `.gql` + tests :+1: